Service-catalog: unable to create binding (http error 500)

When user tried to create binding via openshift webUI, binding is not created.

service-broker log:
```
2018-05-21 06:38:00 INFO  Reflector:55 - Waiting on event queue for 599998 ms unless notified
2018-05-21 06:44:13 INFO  OSBBindingService:50 - Received bind request for instance 2fe1fa23-b117-416f-8d36-2e7a1aab9f62, binding f8efe1af-3c28-4139-8b1a-921c0f3ad511 (service id 1e1b59e2-75bc-3e20-bf7b-ae9e07ecdff3, plan id 1f3ae309-e188-3c2f-b8cd-5ffe8e3cf5ea)
2018-05-21 06:44:14 WARN  DefaultExceptionMapper:53 - Returning server error HTTP status 500
javax.ws.rs.InternalServerErrorException: Exception interacting with auth service
	at io.enmasse.osb.api.bind.OSBBindingService.bindServiceInstance(OSBBindingService.java:119)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:140)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:294)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:248)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:235)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:398)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:205)
	at org.jboss.resteasy.plugins.server.vertx.RequestDispatcher.service(RequestDispatcher.java:93)
	at org.jboss.resteasy.plugins.server.vertx.VertxRequestHandler.lambda$handle$0(VertxRequestHandler.java:65)
	at io.vertx.core.http.HttpServerRequest.lambda$bodyHandler$0(HttpServerRequest.java:209)
	at io.vertx.core.http.impl.HttpServerRequestImpl.handleEnd(HttpServerRequestImpl.java:417)
	at io.vertx.core.http.impl.Http1xServerConnection.handleLastHttpContent(Http1xServerConnection.java:462)
	at io.vertx.core.http.impl.Http1xServerConnection.handleContent(Http1xServerConnection.java:457)
	at io.vertx.core.http.impl.Http1xServerConnection.processMessage(Http1xServerConnection.java:437)
	at io.vertx.core.http.impl.Http1xServerConnection.handleMessage(Http1xServerConnection.java:141)
	at io.vertx.core.http.impl.HttpServerImpl$ServerHandlerWithWebSockets.handleMessage(HttpServerImpl.java:728)
	at io.vertx.core.http.impl.HttpServerImpl$ServerHandlerWithWebSockets.handleMessage(HttpServerImpl.java:636)
	at io.vertx.core.net.impl.VertxHandler.lambda$channelRead$1(VertxHandler.java:146)
	at io.vertx.core.impl.ContextImpl.lambda$wrapTask$2(ContextImpl.java:337)
	at io.vertx.core.impl.ContextImpl.executeFromIO(ContextImpl.java:195)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:144)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1380)
	at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1159)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1194)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:489)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:428)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1359)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:935)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:141)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: javax.ws.rs.NotFoundException: HTTP 404 Not Found
	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.handleErrorStatus(ClientInvocation.java:218)
	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.extractResult(ClientInvocation.java:189)
	at org.jboss.resteasy.client.jaxrs.internal.proxy.extractors.BodyEntityExtractor.extractEntity(BodyEntityExtractor.java:60)
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:107)
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:76)
	at com.sun.proxy.$Proxy50.groups(Unknown Source)
	at io.enmasse.osb.keycloak.KeycloakInstance.getGroups(KeycloakInstance.java:38)
	at io.enmasse.osb.api.bind.OSBBindingService.getAllExistingGroups(OSBBindingService.java:150)
	at io.enmasse.osb.api.bind.OSBBindingService.ensureAllGroups(OSBBindingService.java:138)
	at io.enmasse.osb.api.bind.OSBBindingService.bindServiceInstance(OSBBindingService.java:73)
... 57 more
```

--- a/service-broker/src/main/java/io/enmasse/osb/api/bind/OSBBindingService.java
+++ b/service-broker/src/main/java/io/enmasse/osb/api/bind/OSBBindingService.java
@@ -77,7 +77,7 @@ public class OSBBindingService extends OSBServiceBase {
             this.random.nextBytes(passwordBytes);
             String password = Base64.getEncoder().encodeToString(passwordBytes);
 
-            UserRepresentation userRep = keycloakClient.createUser(addressSpace.getName(), username, password);
+            UserRepresentation userRep = keycloakClient.createUser(addressSpace.getAnnotation(AnnotationKeys.REALM_NAME), username, password);
 
             createGroupMapping(keycloakClient, addressSpace, userRep.getId(), groupIds);
 
@@ -128,7 +128,7 @@ public class OSBBindingService extends OSBServiceBase {
     private void createGroupMapping(KeycloakClient keycloakClient, AddressSpace addressSpace, String userId, List<String> groupIds) throws IOException, GeneralSecurityException  {
         for(String groupId : groupIds) {
 
-            keycloakClient.joinGroup(addressSpace.getName(), userId, groupId);
+            keycloakClient.joinGroup(addressSpace.getAnnotation(AnnotationKeys.REALM_NAME), userId, groupId);
         }
 
     }
@@ -140,14 +140,14 @@ public class OSBBindingService extends OSBServiceBase {
             if(existingGroups.containsKey(group)) {
                 groupIds.add(existingGroups.get(group).get(0).getId());
             } else {
-                groupIds.add(keycloakClient.createGroup(addressSpace.getName(), group).getId());
+                groupIds.add(keycloakClient.createGroup(addressSpace.getAnnotation(AnnotationKeys.REALM_NAME), group).getId());
             }
         }
         return groupIds;
     }
 
     private Map<String, List<GroupRepresentation>> getAllExistingGroups(KeycloakClient keycloakClient, AddressSpace addressSpace) {
-        return keycloakClient.getGroups(addressSpace.getName()).stream()
+        return keycloakClient.getGroups(addressSpace.getAnnotation(AnnotationKeys.REALM_NAME)).stream()
                 .collect(Collectors.groupingBy(GroupRepresentation::getName));
     }
 

